src/main/java/com/google/sps/data/Post.java

```diff
@@ -3,13 +3,20 @@
 import com.google.appengine.api.blobstore.BlobKey;
 
 public final class Post {
+  private long id;
   private String caption;
   private BlobKey blobKey;
   private String creator;
+  private String location;
+  private String placeId;
 
-  public Post(String caption, BlobKey blobKey, String creator) {
+  public Post(long id, String caption, BlobKey blobKey, String creator, 
```

Comment:
The recommended way to break methods with a large number of arguments is as follows: 

```java
public Post(
    long id, 
    String caption, 
    BlobKey blobKey, 
    String creator, 
    String location, 
    String placeId) {
```

The external style guide doesn't show any examples, but we use this internally. 

More generally, we try to avoid constructors that take this many arguments, and instead use the `Builder` pattern. See [this](https://howtodoinjava.com/design-patterns/creational/builder-pattern-in-java/) for an overview. The book "Effective Java" has a great overview of this pattern.

Reply:
Same applies to other code locations in this PR.